src/units.cpp

```diff
@@ -410,11 +410,17 @@ size_t Units::unitCount() const
 double Units::scalingFactor(const UnitsPtr &units1, const UnitsPtr &units2)
 {
     double multiplier = 0.0;
+    double scalingFactor = 0.0;
 
-    updateUnitMultipliers(multiplier, units2, 1, 0, 1);
-    updateUnitMultipliers(multiplier, units1, 1, 0, -1);
+    if (!units1 || !units2) {
```

Comment:
Interesting, I would have expecting `clang-tidy` to complain about this kind of test. If anything, it should have asked for something like `if (units1 == nullptr || units2 == nullptr) {`...!?